Billing.API/Services/Invoice/DummyInvoiceProviderService.cs

```
@@ -7,23 +7,32 @@ namespace Billing.API.Services.Invoice
 {
     public class DummyInvoiceProviderService : IInvoiceService
     {
-        public async Task<IEnumerable<InvoiceListItem>> GetInvoices(string clientId)
+        public async Task<IEnumerable<InvoiceListItem>> GetInvoices(string origin, int clientId)
         {
-            if (clientId.In("000000000000000", "999999999999999"))
-                return await Task.Run(() => new List<InvoiceListItem>());
+            if (clientId <= 0)
+                return null!;
+
+            string clientPrefix;
+
+            if (origin.EqualsIgnoreCase("doppler"))
+                clientPrefix = "CD";
+            else if (origin.EqualsIgnoreCase("relay"))
+                clientPrefix = "CR";
+            else
+                return null!;
```

Again, in my opinion, it should throw an argument exception. Maybe you can move this validation out of the ServiceProvider and deal with these scenarios in the controller or in another service consumed by the controller.

---

Another argument in favor of moving this logic out of this service is that now you have to duplicate this code in DummyInvoiceProviderService and InvoiceService. (BTW, why the dummy one is called _ProviderService_ and the other one is called simply _Service_?)

---

I think we are not considering Client Manager in this version, right?. A Doppler account could be a Client or Client Manager. If the latter is the case, ClientPrefix must be "CM".